src/OmniSharp.Roslyn.CSharp/Services/Diagnostics/CodeCheckService.cs

```diff
@@ -1,34 +1,60 @@
-﻿using System.Collections.Generic;
-using System.Composition;
+﻿using System.Composition;
 using System.Linq;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis;
 using OmniSharp.Helpers;
 using OmniSharp.Mef;
 using OmniSharp.Models;
 using OmniSharp.Models.CodeCheck;
+using System.Collections.Generic;
+using OmniSharp.Models.Diagnostics;
+using Microsoft.Extensions.Logging;
 
 namespace OmniSharp.Roslyn.CSharp.Services.Diagnostics
 {
     [OmniSharpHandler(OmniSharpEndpoints.CodeCheck, LanguageNames.CSharp)]
     public class CodeCheckService : IRequestHandler<CodeCheckRequest, QuickFixResponse>
     {
-        private OmniSharpWorkspace _workspace;
+        private readonly OmniSharpWorkspace _workspace;
+        private readonly RoslynAnalyzerService _roslynAnalyzer;
+        private readonly ILogger<CodeCheckService> _logger;
 
         [ImportingConstructor]
-        public CodeCheckService(OmniSharpWorkspace workspace)
+        public CodeCheckService(OmniSharpWorkspace workspace, RoslynAnalyzerService roslynAnalyzer, ILoggerFactory loggerFactory)
         {
             _workspace = workspace;
+            _roslynAnalyzer = roslynAnalyzer;
+            _logger = loggerFactory.CreateLogger<CodeCheckService>();
         }
 
         public async Task<QuickFixResponse> Handle(CodeCheckRequest request)
         {
-            var documents = request.FileName != null
-                ? _workspace.GetDocuments(request.FileName)
-                : _workspace.CurrentSolution.Projects.SelectMany(project => project.Documents);
+            var projects = !string.IsNullOrEmpty(request.FileName)
+                ? _workspace.GetDocuments(request.FileName).Select(x => x.Project).ToList()
+                : _workspace.CurrentSolution.Projects.ToList();
```

Comment:
Why are the additional heap allocations of `ToList` needed?

Reply:
Leftovers from debugging responses, visual studio has problems to evaluate results during debugging 😞

Reply:
@savpek I think you could be interested in [OzCode extension](https://www.oz-code.com/) for VS, I've started to use it lately. Immense help with LINQ debugging. Disclosure: I'm not affiliated in any way 😅